core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/conditional/DecodeExpression.kt

```diff
@@ -0,0 +1,42 @@
+package ch.ergon.dope.resolvable.expression.unaliased.type.conditional
+
+import ch.ergon.dope.DopeQuery
+import ch.ergon.dope.resolvable.expression.TypeExpression
+import ch.ergon.dope.resolvable.expression.UnaliasedExpression
+import ch.ergon.dope.resolvable.operator.FunctionOperator
+import ch.ergon.dope.validtype.ValidType
+
+class DecodeExpression<T : ValidType, U : ValidType>(
```

Comment:
Comprehension question for this class: Is it correct, that you can look at this Expression kinda like the Conditional Expression i am working on in DOPE-231, but only in a function? 
The searchedResultExpressions are the When-Thens and the default is the else

Reply:
That is my understanding as well. jOOQ even declares `CASE` and `DECODE` [viable alternatives to each other](https://www.jooq.org/doc/latest/manual/sql-building/column-expressions/case-expressions/).